extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -150,7 +150,8 @@ export class BookTocManager implements IBookTocManager {
 		});
 
 		this._tocFiles.forEach(async (value, key) => {
```

Comment:
Are you purposely not awaiting on these tasks to finish? Awaits inside foreachs won't actually be awaited in the calling function - you need to use map and Promise.all for that.